drupal/sites/default/boinc/modules/boincwork/boincwork.module

```diff
@@ -1041,7 +1041,133 @@ function boincwork_get_output() {
  * Get the project configuration XML; used by client software
  */
 function boincwork_get_project_config() {
-  include_boinc('user/get_project_config.php');
+  require_boinc( array('boinc_db', 'util', 'xml') );
+
+  // Begin Output
+  xml_header();
+
+  // get BOINC project config
+  $config = get_config();
+  $long_name = parse_config($config, "<long_name>");
+
+  $min_passwd_length = parse_config($config, "<min_passwd_length>");
+  if (!$min_passwd_length) {
+    $min_passwd_length = 6;
+  }
+  $disable_account_creation = parse_bool($config, "disable_account_creation");
+
+  // Main project config section
+  echo "<project_config>
+    <name>$long_name</name>
+    <master_url>$master_url</master_url>
+    <web_rpc_url_base>".secure_url_base()."</web_rpc_url_base>
+  ";
+
+  if (parse_config($config, "<account_manager>")) {
+    echo "    <account_manager/>\n";
+  }
+  $local_revision = @trim(file_get_contents("../../local.revision"));
+  if ($local_revision) {
+    echo "<local_revision>$local_revision</local_revision>\n";
+  }
+
+  if (web_stopped()) {
+    echo "
+        <error_num>-183</error_num>
+        <web_stopped>1</web_stopped>
+    ";
+  }
+  else {
+    echo "    <web_stopped>0</web_stopped>\n";
+  }
+
+  if ($disable_account_creation || defined('INVITE_CODES')) {
+    echo "    <account_creation_disabled/>\n";
+  }
+
+  if (defined('INVITE_CODES')) {
+    echo "    <invite_code_required/>\n";
+  }
+
+  echo "    <min_passwd_length>$min_passwd_length</min_passwd_length>\n";
+
+  if (sched_stopped()) {
+    echo "    <sched_stopped>1</sched_stopped>\n";
+  }
+  else {
+    echo "    <sched_stopped>0</sched_stopped>\n";
+  }
+
+  $min_core_client_version = parse_config($config, "<min_core_client_version>");
+  if ($min_core_client_version) {
+    echo "<min_client_version>$min_core_client_version</min_client_version>\n";
+  }
+
+  // Display platforms
+  boincwork_show_platforms();
+
+  // Terms of use (from Drupal)
+  $termsofuse = variable_get('boinc_weboptions_termsofuse', '');
```

Comment:
Hm, so `boinc_weboptions_termsofuse` is in fact the actual text? If so, my comment above on RPC coherence is obsolete. Should this be the actual terms of use text, does this completely replace BOINC's stock [terms of use file](https://boinc.berkeley.edu/trac/wiki/TermsOfUse) (also for the BOINC client etc), presumably because you reimplemented the whole `get_project_config` RPC?

Would it be possible to use BOINC's standard terms of use file instead and thereby avoid reimplementing the RPC entirely which would largely facilitate maintenance? Encapsulation and extension would of course be fine.